beancount_interpolate/common.py

```diff
@@ -103,26 +126,26 @@ def distribute_over_period(params, default_date, total_value, config):
     """
 
     begin_date, duration, step = parse_mark(params, default_date, config)
-    period = math.floor( duration / step )
+    period = get_number_of_txn(begin_date, duration, step, max_txn=config['max_new_tx'])
 
     if(period > config['max_new_tx']):
         period = config['max_new_tx']
         duration = period * step
```

Comment:
this block seems redundant now, as you're handling max_new_tx within get_number_of_txn.